### PR TITLE
347 Refactor desktop, mobile and sub navigation

### LINKED ISSUE #347: [347] Refactor Desktop Menu
**Current state**
We use the old menu at the top

image

**Expected state**
We use the new menu according to design

**On a post**
image

**On homepage**
image

Search and filter don't work yet

**Necessities**
#0

**Branch**
347-refactor-desktop-menu

### PR DESCRIPTION
**Pull request by AjobK**
closes #347 

<br />

 **What is it supposed to do**  
The desktop, mobile and sub navigations has been revamped and refactored according to design. A few things differ from the design as noted in the issue, but this is intentional. Please refer to #347.

The scope of the issue was only the desktop navigation, but I did all of the navigations because it needed reworking badly and the inconsistencies were painful to look at.

Also, removed ui and nav store. It is bad practice to put UI related things in a global store like MobX. Only following stores are still available:
image

user store will be removed at another stage in the project (#337)

<br />

 **How can we test your branch**  
- [ ] Check code
- [ ] Click on desktop navigation buttons
- [ ] Check desktop navigation responsiveness
- [ ] Click on hamburger icon
- [ ] Check if subnavigation items are correct
- [ ] Resize to mobile with subnavigation open (responsiveness)
- [ ] Check mobile navigation works correctly
- [ ] Try to log out
- [ ] Click on avatar
- [ ] Spam hamburger?
- [ ] Do random stuff...
- [ ] Use your imagination

<br />

 **Additional notes**  
Things to keep in mind regarding the PR that don't fit in the previous two headings. 

<br />

**1. General checklist**
- [x] a. Tested happy flow.
- [x] b. Tested unhappy flow.
- [x] c. No unexpected exceptions.
- [x] d. No code with high complexity. (Big O)
- [x] e. EOF newline.
- [x] f. No linter warnings or errors.

<br />

**2. If relevant, front-end checklist**
- [x] a. Responsive. (Desktop all the way to mobile 320px)
- [x] b. Using SCSS variables where relevant.
- [x] c. Relevant class names
- [x] d. No modified global SCSS properties, except if necessary and fully tested the impact.
- [x] e. Checked [caniuse](https://caniuse.com) for more modern CSS properties.

<br />

**3. If relevant, back-end checklist**
- [ ] a. No unhandled promise rejections.
- [ ] b. Seeders still functional.
- [ ] c. No primary-key violations.
- [ ] d. Normalization, at least [3NF](https://en.wikipedia.org/wiki/Database_normalization).
- [ ] e. Used guards (middleware) where necessary and tested if route is guarded.

<br />

**4. If relevant, test these browsers**
- [x] a. Chrome
- [x] b. Firefox
- [x] c. Opera
- [x] d. Edge
- [ ] e. Safari

---

**PR Rules**  
*1. Add relevant labels.*  
*2. Select at least two reviewers.*  
*3. Assign all reviewers allowed to merge the branch, including yourself.*  
*4. If relevant to the sprint, put related issue in 'Awaiting Approval' column on project board.*  
*5. Add to the corresponding [milestone](https://github.com/AjobK/Seaqull/milestones).*

<br />

**Reviewer info**  
*1. You can use aliases to refer to specific PR checklist items. For example 'Hey @JohnDoe, for this line please check 3.a and 1.d.'*  
*2. Where possible, please use the suggestions feature in github so the author has a better idea of the solution you had in mind.*   

<br />

> This form should be filled in by the author of this PR.

### CODE REVIEW

**Review by sribadal — DISMISSED** [review]
It looks good, and works as intended

File: frontend/src/components/subNavigation/menuItems.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+    "menuItemsLoggedIn": {
```
Comment (ryankroon00) [inline frontend/src/components/subNavigation/menuItems.json]: Indentation is incorrect

File: frontend/src/layouts/header/index.js [hunk 2]
```diff
@@ -1,38 +1,79 @@
...
-    ui.toggleSubNav()
+    this.state = {
+      subNavOpen: false
+    }
+
+    this.currentPath = this.props.location.pathname
+  }
+
+  toggleSubNavigation = () => {
+    let { subNavOpen } = this.state
```
Comment (ryankroon00) [inline frontend/src/layouts/header/index.js]:
```suggestion
    const { subNavOpen } = this.state
```

File: frontend/src/components/subNavigation/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import * as menuItemsJSON from './menuItems.json'
+
+@inject('store') @observer
+class SubNavigation extends Component {
+  render() {
+    const { profile } = this.props.store
+    const { className } = this.props
+
+    let menuItems = profile.loggedIn && menuItemsJSON.menuItemsLoggedIn || menuItemsJSON.menuItemsLoggedOut
+    let navigationItems = []
```
Comment (ryankroon00) [inline frontend/src/components/subNavigation/index.js]:
```suggestion
    const menuItems = profile.loggedIn && menuItemsJSON.menuItemsLoggedIn || menuItemsJSON.menuItemsLoggedOut
    const navigationItems = []
```

Comment (ryankroon00) [inline frontend/src/components/subNavigation/index.js]: Maybe change navigationItems to navigationElements to make it easier to understand the difference between menuItems and navigationItems

**Review by ryankroon00 — CHANGES_REQUESTED** [review]
Good work, just a few small suggestions <3

File: frontend/src/components/headerNavigation/index.js [hunk 4]
```diff
@@ -3,31 +3,67 @@ import { observer, inject } from 'mobx-react'
...
+        <div className={ styles.goBackWrapper }>
+          <Link to='/' className={ styles.goBack }>
+            <Icon iconName='Home' className={ styles.goBackIcon } />
+            <p>
+              <span>back to </span>home
+              <span className={ styles.underline } />
+            </p>
+          </Link>
+        </div>
+        <div className={ styles.rightContent }>
```
Comment (AjobK) [inline frontend/src/components/headerNavigation/index.js:33]: Done by moving Logo out of it

**Review by ryankroon00 — APPROVED** [review]

**Pull request merged**